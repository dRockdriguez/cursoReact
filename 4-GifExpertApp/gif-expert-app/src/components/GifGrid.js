import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifDetail from "./GifDetail";
import 'animate.css';
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3> 
      {loading ? (
        <p>Cargando...</p>
      ) : (
        <div className="animate__animated animate__fadeIn">
          <ol className="card-grid">
            {gifs.map((img) => (
              <GifDetail key={`${img.id}${Date.now()}`} {...img} />
            ))}
          </ol>
        </div>
      )}
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid;
