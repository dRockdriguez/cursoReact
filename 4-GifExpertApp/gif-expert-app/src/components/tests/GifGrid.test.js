import React from "react";
import GifGrid from "../GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("GifGrid", () => {
  const category = "pikachu";
  
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("Should render the component", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Should render items when call to api (useFetchGifs)", () => {
    const gifs = [
      {
        id: "1",
        title: "title",
        url: "url",
      },
      {
        id: "2",
        title: "title",
        url: "url",
      }
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifDetail').length).toBe(gifs.length)
  });
});
