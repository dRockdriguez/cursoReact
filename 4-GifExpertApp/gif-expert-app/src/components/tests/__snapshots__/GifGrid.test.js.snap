// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Should render items when call to api (useFetchGifs) 1`] = `
<Fragment>
  <h3>
    pikachu
  </h3>
  <div
    className="animate__animated animate__fadeIn"
  >
    <ol
      className="card-grid"
    >
      <GifDetail
        id="1"
        key="11645716702649"
        title="title"
        url="url"
      />
      <GifDetail
        id="2"
        key="21645716702649"
        title="title"
        url="url"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`GifGrid Should render the component 1`] = `
<Fragment>
  <h3>
    pikachu
  </h3>
  <p>
    Cargando...
  </p>
</Fragment>
`;
